### DISCUSSION TITLE
Is a Primaza worker cluster similar to a vcluster ?

### DISCUSSION THREAD

**Discussion by waveywaves** [primaza/primaza#228] [General]
Based on the architecture it initially seemed like primiza might be spinning up a vcluster https://github.com/loft-sh/vcluster. But doesn't seem like it's the case. What is primaza using to back up the worker cluster ? Also, can a user choose which worker cluster they would like to use eg: k3s, k8s, k0s ?

**Comment by filariow** [comment]
Hi @waveywaves, thank you for reaching out!

Primaza is requiring clusters to already exist and to have communication set-up.
Technically it is ignoring which technology is behind the Kubernetes API, so it should work with all the projects you listed.

Please take a look at our documentation on how to create a [Multi-Cluster environment](https://www.primaza.io/tutorials/tenant/multi-cluster.html).
The tutorial is relying on kind clusters.

We didn't try yet with virtual clusters, but it is something we are planning to do.
